webapp/app/js/compareperf.js

```diff
@@ -133,25 +134,20 @@ perf.controller('CompareResultsCtrl', [
 
     //TODO: duplicated in comparesubtestctrl
     function verifyRevision(project, revision, rsid) {
-      var uri = thServiceDomain + '/api/project/' + project.name +
-          '/resultset/?full=false&revision=' +
-          revision;
-
-      return $http.get(uri).then(function(response) {
-        var results = response.data.results;
-        if (results.length > 0) {
+      return ThResultSetModel.getResultSetsFromRevision(project.name, revision).then(
+        function(resultSets) {
+          var resultSet = resultSets[0];
           //TODO: this is a bit hacky to pass in 'original' as a text string
           if (rsid == 'original') {
-            $scope.originalResultSetID = results[0].id;
-            $scope.originalTimestamp = results[0].push_timestamp;
+            $scope.originalResultSetID = resultSet.id;
+            $scope.originalTimestamp = resultSet.push_timestamp;
           } else {
-            $scope.newResultSetID = results[0].id;
-            $scope.newTimestamp = results[0].push_timestamp;
+            $scope.newResultSetID = resultSet.id;
+            $scope.newTimestamp = resultSet.push_timestamp;
           }
-        } else {
-          $scope.errors.push("No results found for revision " + revision + " on branch " + project);
-        }
-      });
+        }, function(error) {
+          $scope.errors.push(error);
```

Comment:
what error is pushed here?  is this an error from getResultSetsFromRevision?  if so, could we see an example?  I would like to ensure that this is friendly for the end user.